sensor/DNSProxyPlugin.js

```diff
@@ -299,21 +314,40 @@ class DNSProxyPlugin extends Sensor {
           }
           item[k] = JSON.stringify(v);
         }
-        
-        await intelTool.addDomainIntel(dn, item, item.e || defaultExpireTime);
-        await this.updateRedisCacheFromIntel(dn, item);
+       
+        if (item.c === "intel" && !skipped) {
+          const deviceips = await hostTool.getIPsByMac(mac.toUpperCase())
+          const deviceip = deviceips[0]
+          const alarm = new Alarm.IntelAlarm(new Date() / 1000, deviceip, "major", {
+            "p.device.ip": deviceip,
+            "p.dest.name": dn,
+            "p.security.reason": item.msg,
+            "p.device.mac": mac,
+            "p.blockby": "fastdns"
+          });
+          await am2.enrichDeviceInfo(alarm)
+          am2.enqueueAlarm(alarm);
+          skipped = true
+
+          await intelTool.addDomainIntel(dn, item, item.e || defaultExpireTime);
+          await rclient.zaddAsync(passthroughKey, Math.floor(new Date() / 1000), dn);
```

Comment:
* this is too soon, it may take time to process the alarm and add the rule. so need to wait for a while to add to zsets.
* another option is to have a callback from *adding rule* logic in alarm manager, when there is a callback or event from there, add to zset. this can make sure the rule is added before adding to zset.